### PR TITLE
[TT-309] Feature/selection

### PR DESCRIPTION
**Pull request by tbuchaillot**
**Removed the -a and -p** shortcuts and now it only accepts **`--apis` and `--policies`.** for selection. 

**Change the dump selection behavior:**
- If one or more APIs are selected, dump those APIs.
- If one or more Policies are selected, dump those Policies.
- If neither Policies nor APIs is specified, dump everything.

For example:
```
dump --apis="api_id_1" <- dumps only the API 1
dump --policies="pol_1" <- dumps only the POL 1
dump --apis="api_id_1"  --policies="pol_1"  <- dumps API 1 and POL 1
dump <- dump everything

```
Extended the selection support to **sync, publish, and update** commands.

### CODE REVIEW

File: cmd/dump.go [hunk 1]
```diff
@@ -56,33 +57,63 @@ var dumpCmd = &cobra.Command{
...
+		wantedAPIs , _ := cmd.Flags().GetStringSlice("apis")
+
 
 		policies := []objects.Policy{}
+		apis := []objects.DBApiDefinition{}
 		var errPoliciesFetch error
+		var errApisFetch error
+
+
+		if len(wantedAPIs) > 0 {
```
Comment (matiasinsaurralde) [inline cmd/dump.go:69]: I think it's safe to remove this check, the iteration should be enough and nothing happens if it's empty.

File: cmd/dump.go [hunk 2]
```diff
@@ -56,33 +57,63 @@ var dumpCmd = &cobra.Command{
...
+			for _, APIID := range wantedAPIs{
+				api :=  objects.DBApiDefinition{}
+				api.APIID = APIID
+				apis = append(apis, api)
+			}
+
+		}
+		if len(wantedPolicies) > 0  {
+			for _,wantedPolicy := range wantedPolicies{
```
Comment (matiasinsaurralde) [inline cmd/dump.go:78]: Same as previous comment.

File: cmd/dump.go [hunk 3]
```diff
@@ -56,33 +57,63 @@ var dumpCmd = &cobra.Command{
...
+				if !bson.IsObjectIdHex(wantedPolicy){
+					fmt.Println("Invalid selected Policiy ID:",wantedPolicy,".")
```
Comment (matiasinsaurralde) [inline cmd/dump.go:80]: Could be better to use: `fmt.Printf("Invalid selected policy ID: %s.\n", wantedPolicy)`

File: cmd/dump.go [hunk 4]
```diff
@@ -56,33 +57,63 @@ var dumpCmd = &cobra.Command{
...
+				pol := objects.Policy{
+					ID: wantedPolicy,
+					MID: bson.ObjectIdHex(wantedPolicy),
+				}
+				policies = append(policies,pol)
+			}
+		}
+
+
+		if len(wantedAPIs) == 0 && len(wantedPolicies) == 0 {
```
Comment (matiasinsaurralde) [inline cmd/dump.go:92]: This check looks fine and it's different from the previous ones that don't log anything, we can keep it.

**Review by matiasinsaurralde — APPROVED** [review]
Looks good, left some minor suggestions.

File: cmd/dump.go (hunk 1 above)
Comment (tbuchaillot) [reply cmd/dump.go]: solved

File: cmd/dump.go (hunk 3 above)
Comment (tbuchaillot) [reply cmd/dump.go]: solved

File: cmd/dump.go (hunk 2 above)
Comment (tbuchaillot) [reply cmd/dump.go]: solved

**Pull request merged**